src/libmongoc/CMakeLists.txt

```diff
@@ -856,35 +856,67 @@ if (ENABLE_SHM_COUNTERS STREQUAL "ON")
    )
 endif ()
 
+cmake_dependent_option (
+   MONGOC_TEST_USE_CSFLE "If TRUE, download and use a csfle dynamic library for client-side-encryption testing. Otherwise will rely on mongocryptd"
+   TRUE ENABLE_TESTS FALSE
+)
+
+if (MONGOC_TEST_USE_CSFLE)
+   find_program (_PYTHON3_EXE NAMES python3 py python)
+   mark_as_advanced (_PYTHON3_EXE)
+   if (NOT _PYTHON3_EXE)
+      message (SEND_ERROR
+         "MONGOC_TEST_USE_CSFLE is enabled, but we were unable to find a Python 3 interpreter to use mongodl.py. "
+         "Install a Python 3 or set MONGOC_TEST_USE_CSFLE to FALSE"
+      )
+   endif ()
+
+   get_filename_component (csfle_dest "${CMAKE_CURRENT_BINARY_DIR}/_csfle/mongo_csfle_v1${CMAKE_SHARED_LIBRARY_SUFFIX}" ABSOLUTE)
```

Comment:
Confirming, is this line setting `CMAKE_SHARED_LIBRARY_SUFFIX`? Otherwise, where and how is `CMAKE_SHARED_LIBRARY_SUFFIX` defined?

Reply:
That's a built-in variable defined by CMake (or a toolchain file) that tells you the default file extension of runtime libraries (e.g. `.so`, `.dll`, or `.dylib`)